src/emitter.c

```diff
@@ -1923,7 +1923,14 @@ yaml_emitter_write_plain_scalar(yaml_emitter_t *emitter,
 
     STRING_ASSIGN(string, value, length);
 
-    if (!emitter->whitespace) {
+    if (!emitter->whitespace && (length || emitter->flow_level)) {
+        /**
+         * Avoid trailing spaces for empty values in block style.
+         * In flow style, the emitter forbids empty values everywhere,
+         * and outputs a single quoted scalar instead.
+         * If this is ever changed, checking for emitter->flow_level
+         * will prevent ambiguous things like {a:}
```

Comment:
@perlpunk, how about...
```
If the emitter is ever changed to allow emitting empty values, then checking for emitter->flow_level
will prevent emitting ambiguous things like {a:}.
```
(with appropriate line wrapping of course :)

Reply:
I rephrased it a bit more and pushed.